extension.js

```diff
@@ -44,16 +51,54 @@ export default class ForceQuitExtension extends Extension {
     enable() {
         this._button = new ForceQuitButton(this.path);
 
-        // Uncomment following line to get it to the left panel besides AppMenu
-        // Main.panel.addToStatusArea(ButtonName, this._button, 3, 'left');
-        Main.panel.addToStatusArea(ButtonName, this._button);
+        // Apply initial state and listen for settings changes
+        this._settings = this.getSettings();
+        this._updateButtonVisibility();
+        this._settingsChangedId = this._settings.connect('changed::hide-button',
+            () => this._updateButtonVisibility()
+        );
+  
+        // Position the button on the top panel
+        if (this._settings.get_string('button-position') == 'left')
+            Main.panel.addToStatusArea(ButtonName, this._button, 3, 'left');
+        else
+            Main.panel.addToStatusArea(ButtonName, this._button);
+
+        // Redraw button on position change
+        this._settings.connect('changed::button-position', () => {
+            this.disable();
+            this.enable();
+        });
+
+        // Export DBus interface
+        this._dbus = Gio.DBusExportedObject.wrapJSObject(DBusInterface, this);
+        this._dbus.export(Gio.DBus.session, '/org/gnome/Shell/Extensions/ForceQuit');
     }
 
     disable() {
+        if (this._settingsChangedId) {
```

Comment:
Q: Why does `disable` here put all these in `if` but the other methods dont? Would the other methods also not be unsafe if the extension is disabled? Say if somehow `_updateButtonVisibility` was called after `disable`, there would be no `this._button` right? Is that a case extensions handle when supporting preferences? If not, or if it is handled by the runtime, why do the if-null here?